storage/badger/protocol_state_test.go

```diff
@@ -234,10 +234,10 @@ func assertRichProtocolStateValidity(t *testing.T, state *flow.RichProtocolState
 		// next epoch's `Participants` [IdentitySkeletons] and `ActiveIdentities` [DynamicIdentity properties] list the same nodes in canonical ordering.
 		participantsFromNextEpochSetup, err = flow.ComposeFullIdentities(state.NextEpochSetup.Participants, state.NextEpoch.ActiveIdentities)
 		assert.NoError(t, err, "should be able to reconstruct next epoch active participants")
-		allIdentities = participantsFromCurrentEpochSetup.Union(participantsFromNextEpochSetup.Map(mapfunc.WithWeight(0)))
+		allIdentities = participantsFromCurrentEpochSetup.Union(participantsFromNextEpochSetup.Map(mapfunc.WithEpochParticipationStatus(flow.EpochParticipationStatusJoining)))
```

Comment:
to prevent accidental modification of underlying Identities, I would suggest to make a copy everywhere where we modify Identities in place: 
```suggestion
		allIdentities = participantsFromCurrentEpochSetup.Union(participantsFromNextEpochSetup.Copy().Map(mapfunc.WithEpochParticipationStatus(flow.EpochParticipationStatusJoining)))
```

Reply:
Generally `Map` copies most of the data(except reference types such as staking/networking pubkeys), but sure, we can do it.